### DISCUSSION TITLE
Run .r script as part of quarto document

### DISCUSSION THREAD

**Discussion by fgsconsulting** [quarto-dev/quarto-cli#9360] [Q&A]
### Description

I'm a total beginner when it comes to the r/quarto world. I have some individual .r scripts that I want to include in a quarto document. I prefer to link the quarto document rather than copy and paste the code in code chunks within the document. How do I point a quarto document to a specific .r script? I understand that there needs to be a YAML section included in the .r script. I'm wanting to know the syntax needed in the .r file as well as the .qmd file to make sure the .qmd accurately retrieves and runs the r file. 

I am running the latest version of quarto. 1.4.553.

Thanks so much for your help!

**Comment by cderv** [comment]
I see two solutions: 

- Either you leverage the script rendering by following this doc and related **knitr** doc as you are using R for computation: https://quarto.org/docs/computations/render-scripts.html

- Or you write a .qmd file directly that will source code in cell from external R file. 
	- See https://bookdown.org/yihui/rmarkdown-cookbook/source-script.html
	- or https://bookdown.org/yihui/rmarkdown-cookbook/option-code.html
	
  depending on what you aim at 

hope this helps
